fix: improve the documentation for cluster resources

jiaqiluo opened the pull request:
<!--- If there is no user issue related to this then you should remove the next line --->
Addresses  
- https://github.com/rancher/terraform-provider-rancher2/issues/1804
- https://github.com/rancher/terraform-provider-rancher2/issues/1738

<!--- Add labels (eg. release/v13) for each release branch to target --->
<!--- Labels need to be added before PR is created for automation to run smoothly! --->

## Description

This PR adds the following missing information to the documentation:

1. Clarification on when to use the `rancher2_cluster` resource versus the `rancher2_cluster_v2` resource.
2. A notice that RKE1 clusters have been deprecated.
3. An example demonstrating how to enable and configure the version-management feature on an imported cluster.
4. Remove examples for creating RKE1 node-driver or custom clusters


## Testing
 
Not a breaking change. Need to ensure the documentation is displayed properly after release.  

----

**Note to reviews**

- The failure of the validate-commit-message workflow can be ignored

HarrisonWAffel commented on docs/resources/cluster.md:
```diff
@@ -4,19 +4,52 @@ page_title: "rancher2_cluster Resource"
 **Note optional/computed arguments** If any `optional/computed` argument of this resource is defined by the user, removing it from tf file will NOT reset its value. To reset it, let its definition at tf file as empty/false object. Ex: `cloud_provider {}`, `name = ""`
 
-### Creating Rancher v2 imported cluster
+### Creating Rancher v2 imported cluster and retrieve registration commands
```
```suggestion
### Creating a Rancher v2 imported cluster and retrieving the registration commands
```

HarrisonWAffel commented on docs/resources/cluster.md:
```diff
@@ -4,19 +4,52 @@ page_title: "rancher2_cluster Resource"
 # rancher2\_cluster Resource
 
-Provides a Rancher v2 Cluster resource. This can be used to create Clusters for Rancher v2 environments and retrieve their information.
+Provides a Rancher v2 Cluster resource. This can be used to create imported Clusters for Rancher v2 environments and retrieve their information.
```
It looks like this file also states it can be used to create [custom clusters](https://github.com/jiaqiluo/terraform-provider-rancher2/blob/update-doc/docs/resources/cluster.md#creating-rancher-v2-rke-cluster)? We should remove the other RKE1 examples from this file if we want to say that only imported clusters are supported - they don't work anymore anyway iiuc.

jiaqiluo replied:
Good point. 
All examples for creating RKE1 node-driver or custom clusters are removed from the doc.

HarrisonWAffel reviewed (changes requested):
minor nits, otherwise lgtm!

jiaqiluo committed:
Improve the documentation for rancher2_cluster and rancher2_cluster_v2 resources

- add examples for configuring the version-management feature on imported cluster
- remove examples for creating RKE1 node-driver or custom clusters

matttrach committed:
Merge branch 'main' into update-doc
